components/constellation/constellation.rs

```diff
@@ -1795,52 +1801,58 @@ where
-                None => {
-                    return warn!(
-                        "Script loaded url in closed iframe {}.",
-                        load_info.info.parent_pipeline_id
-                    )
-                },
+            let window_size = self.browsing_contexts
+                .get(&browsing_context_id)
+                .and_then(|ctx| ctx.size);
+
+            let is_parent_private = {
+                let parent_browsing_context_id =
+                    match self.pipelines.get(&parent_pipeline_id) {
+                        Some(pipeline) => pipeline.browsing_context_id,
+                        None => return warn!(
+                            "Script loaded url in iframe {} in closed parent pipeline {}.",
+                            browsing_context_id,
+                            parent_pipeline_id,
+                        ),
+                    };
+                let is_parent_private =
+                    match self.browsing_contexts.get(&parent_browsing_context_id) {
+                        Some(ctx) => ctx.is_private,
+                        None => return warn!(
+                            "Script loaded url in iframe {} in closed parent browsing context {}.",
+                            browsing_context_id,
+                            parent_browsing_context_id,
+                        ),
+                    };
+                is_parent_private
             };
-            let (is_parent_private, is_parent_visible) =
-                match self.browsing_contexts.get(&parent_browsing_context_id) {
-                    Some(browsing_context) => (
-                        browsing_context.is_private,
-                        browsing_context.is_visible,
-                    ),
-                    None => {
-                        return warn!(
-                            "Script loaded url from closed browsing context {}.",
-                            parent_browsing_context_id
-                        )
-                    },
-                };
-            let is_private = load_info.info.is_private || is_parent_private;
-
-            let (window_size, is_visible) = self.browsing_contexts
-                .get(&load_info.info.browsing_context_id)
-                .map(|context| (context.size, context.is_visible))
-                .unwrap_or((None, is_parent_visible));
+            let is_private = is_private || is_parent_private;
 
-            (load_data, window_size, is_private, is_visible)
+            (load_data, window_size, is_private)
         };
 
-        let replace = if load_info.info.replace {
-            self.browsing_contexts
-                .get(&load_info.info.browsing_context_id)
-                .map(|browsing_context| NeedsToReload::No(browsing_context.pipeline_id))
-        } else {
-            None
+        let (replace, is_visible) = {
+            let browsing_context = match self.browsing_contexts.get(&browsing_context_id) {
```

Comment:
Can theses browsing context lookups be combined with the ones from above?

Reply:
Ah, good catch, I overlooked that. On it!